### DISCUSSION TITLE
Windows EAP Client connects but doesn't leave the server

### DISCUSSION THREAD

**Discussion by CBB3810** [strongswan/strongswan#1654] [Q&A] [answered]
`conn base
  keyexchange=ikev2
  auto=add
  compress=no
  type=tunnel
  fragmentation=yes
  forceencaps=yes
  dpdaction=clear
  dpddelay=120s
  #mobike=yes
  eap_identity=%identity
  reauth=no
  rekey=no

conn joynd-srvr
#Server Side Configs
  left=%any
  leftid=10.0.12.30
  leftsubnet=10.0.12.0/24,10.0.11.0/24,10.0.15.0/24,10.0.0.2/32
  leftcert=server-cert.pem
  leftsendcert=always
  also=base

conn joynd-client
#Client Side Configs
  right=%any
  rightid=%any
  rightauth=eap-mschapv2
  rightsourceip=100.64.100.0/24
  rightdns=10.0.0.2,8.8.4.4
  rightsendcert=never
  also=base`

I get connected, I get an ip address of 100.64.100.1/24 and I can ping the VPN ens5 internface IP of 10.0.12.30.
Beyond that my packets won't go anywhere. I'm stumped.

**Comment by drcyberg** [comment]
Hi!
I think you want to route the network traffic through the remote machine. 
If so, the next parameter must be changed to `rightsubnet=0.0.0.0/0`
You can find more information here: [LINK](https://wiki.strongswan.org/projects/strongswan/wiki/IKEv2ClientConfig)

**Comment by tobiasbrunner** [comment]
> leftsubnet=10.0.12.0/24,10.0.11.0/24,10.0.15.0/24,10.0.0.2/32

If the problem is that the client only sends traffic for `10.0.12.0/24` through the tunnel but not e.g. for `10.0.11.0/24`, then what clients are you using? Not all clients support multiple traffic selectors like this. **Edit:** Missed the title, so also see [the docs regarding split-tunneling](https://docs.strongswan.org/docs/5.9/howtos/forwarding.html#_split_tunneling).

If the problem is that traffic is arriving at the server but not forwarded, then please have a look [at the docs](https://docs.strongswan.org/docs/5.9/howtos/forwarding.html).

**Comment by CBB3810** [comment]
I apologize if this is too much info as a response. 
First I'd like to say that I appreciate the responses. Here's my feedback and I also attached a connection log if it's helpful:
[ConnectionLog.txt](https://github.com/strongswan/strongswan/files/11233380/ConnectionLog.txt)
** I'm unclear about this** -- Where do I define local_ts and remote_ts? 
Hosts on the Internet
If split-tunneling is not used, all client traffic will be sent through the IPsec tunnel. 
In this scenario 'local_ts = 0.0.0.0/0` is configured on the gateway and remote_ts = 0.0.0.0/0 on the client.

I've provided everything else I could imagine would be relevant. 
Please let me know if there's anything else I can supply that would be of use.
As stated previously, vpn connects. I just can't ping past the server side.

StrongSwan Server version: strongSwan 5.9.5 Installed natively through Ubuntu 22.04 pkg manager.
sudo apt install -y strongswan strongswan-pki libcharon-extra-plugins libcharon-extauth-plugins libstrongswan-extra-plugins libtss2-tcti-tabrmd-dev

Sysctl changes that are already accomplished:
net.ipv4.ip_forward = 1
net/ipv4/conf/all/accept_redirects=0
net.ipv4.conf.all.send_redirects = 0
net/ipv4/ip_no_pmtu_disc=1

(UFW firewall is deployed) Here are my relevant /etc/ufw/before.rules changes
*nat
# Allow outbound traffic from 100.64.100.0/24 for specific
-A POSTROUTING -s 100.64.100.0/24 -o ens5 -m policy --pol ipsec --dir out -j ACCEPT
-A POSTROUTING -s 100.64.100.0/24 -o ens5 -j MASQUERADE
COMMIT

*mangle
-A FORWARD --match policy --pol ipsec --dir in -s 100.64.100.0/24 -o ens5 -p tcp -m tcp --tcp-flags SYN,RST SYN -m tcpmss --mss 1461:1536 -j TCPMSS --set-mss 1460
-A FORWARD --match policy --pol ipsec --dir out -s 100.64.100.0/24 -o ens5 -p tcp -m tcp --tcp-flags SYN,RST SYN -m tcpmss --mss 1461:1536
 -j TCPMSS --set-mss 1460
COMMIT

-A ufw-before-forward --match policy --pol ipsec --dir in --proto esp -s 100.64.100.0/24 -j ACCEPT
-A ufw-before-forward --match policy --pol ipsec --dir out --proto esp -d 100.64.100.0/24 -j ACCEPT
#-A ufw-before-forward -s 100.64.100.0/24 -m state --state NEW,RELATED,ESTABLISHED -j ACCEPT  << I tried adding this but it didn't seem to help anything.

On windows client as additional info: I did: 
Add-VpnConnection -Name "Strongswantest03" -ServerAddress "host.fqdn.redacted.com" -TunnelType "IKEv2" -EncryptionLevel "Maximum" -AuthenticationMethod Eap -PassThru -Force
Set-VpnConnectionIPsecConfiguration -Name "Strongswantest03" -AuthenticationTransformConstants GCMAES256 -CipherTransformConstants GCMAES256 -DHGroup ECP384 -IntegrityCheckMethod SHA384 -PfsGroup ECP384 -EncryptionMethod GCMAES256 -Force
Set-VpnConnection -Name "Strongswantest03" -SplitTunneling 0

I also tried manually adding routes to the client's tunnel:
Add-VpnConnectionRoute -ConnectionName "Strongswantest03" -DestinationPrefix "10.0.11.0/24" -PassThru  <<-- this did add an extra persistent route when the vpn tunnel is brought up but didn't help my issue.

TIA to everyone who took the time to look.

**Reply by CBB3810** [reply]
I actually added the rightsubnet=0.0.0.0/0 and it made my ssh session to the host freeze while connected with the client which was super weird and I didn't understand that either. I commented out that part, restarted strongswan and was able to reconnect right away.

**Reply by tobiasbrunner** [reply, marked as answer]
> ** I'm unclear about this** -- Where do I define local_ts and remote_ts?

You are using the legacy config file (ipsec.conf), the documentation refers to swanctl.conf. The parameters are named differently, `local_ts` is `leftsubnet`  while `remote_ts` is `rightsubnet` (see e.g. [here for help on mapping them](https://wiki.strongswan.org/projects/strongswan/wiki/Fromipsecconf)).

> I've provided everything else I could imagine would be relevant.

According to the log, the local traffic selector is just `10.0.12.30/32` (see `proposing traffic selectors for us` in the log). And you also see why if you look at the connection selection:
```
Apr 14 13:31:56 ip-10-0-12-30 ipsec[4972]: 11[CFG] selected peer config 'base'
Apr 14 13:31:56 ip-10-0-12-30 ipsec[4972]: 11[IKE] peer requested EAP, config unacceptable
Apr 14 13:31:56 ip-10-0-12-30 ipsec[4972]: 11[CFG] switching to peer config 'joynd-srvr'
Apr 14 13:31:56 ip-10-0-12-30 ipsec[4972]: 11[IKE] peer requested EAP, config unacceptable
Apr 14 13:31:56 ip-10-0-12-30 ipsec[4972]: 11[CFG] switching to peer config 'joynd-client'
```
Either change `leftsubnet` for the `joynd-client` connection or change the authentication settings for the `joynd-srvr` connection (or simplify the whole config if you only need a single connection).

> I actually added the rightsubnet=0.0.0.0/0

That's definitely wrong as you don't want to send all traffic to your client. When using virtual IPs, don't set `rightsubnet`. If you want all traffic from the client to be sent to the server (unless you possibly override that on the client via `Add-VpnConnectionRoute`), you could configure `leftsubnet=0.0.0.0/0`.

**Comment by CBB3810** [comment]
Tobias, thank you!
EUREKA! It's working!! My understanding of the configs based on the documentation is difficult to conceptualize because I'm an old network engineer familiar with CISCO, PaloAlto etc.. This is my first "rodeo" into Linux VPN hosts. I really appreciate the pointer. This is really cool!

**Comment by CBB3810** [comment]
Last question.. RE: /etc/ipsec.conf directive rightauth=eap-mschapv2
Can I add to the rightauth for 2FA?:
rightauth=eap-mschapv2, radius

Then add a radius cloud config like:
radius_server {
  accounting = {
    server = us1.radius.jumpcloud.com,us2.radius.jumpcloud.com
    secret = radius_secret
    port = 1813
    type = accounting
  }
  authentication = {
    server = us1.radius.jumpcloud.com,us2.radius.jumpcloud.com
    secret = radius_secret
    port = 1812
    type = authentication
    eap = ttls
    identity = %identity  
    password = %password
    pap {
      send_secret = yes
      with_ntdomain_hack = no
    }
    auth_header = '%password,%otp'
  }
}

**Reply by tobiasbrunner** [reply]
> Can I add to the rightauth for 2FA?:

In what way would that be 2FA? Anyway, you can't configure multiple authentication methods in `rightauth`. While you could configure multiple authentication rounds via `rightauth2`, that extension (RFC 4739) is not widely supported (basically only strongSwan supports it).

**Reply by CBB3810** [reply]
Thanks Tobias.
